src/Couchbase.Extensions.MultiOp/MultiOpCollectionExtensions.Get.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using Couchbase.Extensions.MultiOp.Internal;
+using Couchbase.KeyValue;
+
+namespace Couchbase.Extensions.MultiOp
+{
+    /// <summary>
+    /// Extensions to perform multiple operations in parallel on <see cref="CouchbaseCollection"/>.
+    /// </summary>
+    public static partial class MultiOpCollectionExtensions
+    {
+        /// <summary>
+        /// Get multiple documents.
+        /// </summary>
+        /// <param name="collection">Collection to use.</param>
+        /// <param name="keys">Keys to get.</param>
+        /// <param name="multiOpOptions">Options to control parallelization.</param>
+        /// <returns>An observable which receives results as they arrive.</returns>
+        /// <remarks>
+        /// Operations are not executed until the observable is subscribed.
+        /// </remarks>
+        public static IObservable<MultiOpResult<IGetResult>> Get(this ICouchbaseCollection collection,
```

Comment:
Yeah, I was one of the people who insisted on Async because it gives a clear indication of the method's design. However, I think in this case it actually causes more confusion. It gives the impression that this method returns `Task<T>`, which it doesn't. Worse, it will encourage users to `await` the return value. If the IObservable<T> is awaited, then it will wait for all operations to complete but then only return the value of the last operation, dropping all others. I can see this being very confusing.

Perhaps we should consider other method names instead? Like `MultiGet` instead of `Get`? Do you have any other ideas?

Reply:
Gotcha, I recall we discussed this before (were not awaiting a Task, its an observable). Yeah, I think MultiGet over Get is better, we handle the "Async" potential confusion in docs.